modules/asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/persistence/SnapshotDto.java

```diff
@@ -181,6 +183,13 @@ public boolean removeAsset(AssetDto asset) {
   }
 
   public Snapshot toSnapshot() {
+    MediaPackage mediaPackage = Conversions.toMediaPackage(mediaPackageXml);
+    // ensure elements are tagged `archive`
+    for (MediaPackageElement element: mediaPackage.getElements()) {
+      if (!Arrays.asList(element.getTags()).contains("archive")) {
+        element.addTag("archive");
+      }
```

Comment:
@lkiesow can you add an INFO or WARN debug line here to identify which archived elements were not previously tagged for archive. It will help debug how they got there in the first place.

Reply:
done.